### PR TITLE
Adding server verification to single purchases and subscriptions 

### PR DESCRIPTION
**Pull request by RafaelHiguera**

### CODE REVIEW

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs [hunk 1]
```diff
@@ -49,24 +49,22 @@ public static void LoadGameOnline()
...
     }
 
-    public static ServerResponseModel verifyAndSaveUserPurchase(string purchaseToken)
+    public static void verifyAndSaveUserPurchase(Product product)
     {
         var values = new Dictionary<string, string>
         {
-            ["purchaseToken"] = purchaseToken
+            ["receipt"] = getReceiptDetails(product.receipt),
+            ["isSubscription"] = product.definition.id.Contains("subscription").ToString()
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs]: Have this check at server, do not do it at client. Just send receipt is good enough.

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs [hunk 2]
```diff
@@ -49,24 +49,22 @@ public static void LoadGameOnline()
...
     {
-        var values = new Dictionary<string, string>
-        {
-            ["packageName"] = packageName,
-            ["productId"] = productId,
-            ["purchaseToken"] = purchaseToken
-        };
-        return sendUnityWebRequest(values, VERIFY_PURCHASE_WITH_API_URL);
+        receipt = String.Join("", receipt.Split('\\'));
+        receipt = receipt.Remove(0, receipt.IndexOf("orderId") - 2);
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs]: Don't make change to it at client. What if you feel you need extra info? Just send all info to server.

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs [hunk 3]
```diff
@@ -118,73 +118,84 @@ public void OnInitializeFailed(InitializationFailureReason error)
     }
 
     public PurchaseProcessingResult ProcessPurchase(PurchaseEventArgs args)
+    {
+#if ONLINE
+        NetworkRequestController.verifyAndSaveUserPurchase(args.purchasedProduct);
+        return PurchaseProcessingResult.Pending;
+#else
+        ExecutePurchase(args.purchasedProduct.definition.id);
+#endif
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs]: Move #endif after line 132

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs [hunk 4]
```diff
@@ -118,73 +118,84 @@ public void OnInitializeFailed(InitializationFailureReason error)
...
+        // Return a flag indicating whether this product has completely been received, or if the application needs
+        // to be reminded of this purchase at next app launch. Use PurchaseProcessingResult.Pending when still
+        // saving purchased products to the cloud, and when that save is delayed.
+        return PurchaseProcessingResult.Complete;
+    }
+
+    public static void ConfirmPendingPurchase(Product product, bool succes)
+    {
+        if (succes)
+            ExecutePurchase(product.definition.id);
```
Comment (Qing451800) [inline GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs]: if (succes)
{
            ExecutePurchase(product.definition.id);
}

Comment (RafaelHiguera) [reply GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs]: Yep

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs (hunk 3 above)
Comment (RafaelHiguera) [reply GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/PurchaseController.cs]: Will do

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs (hunk 2 above)
Comment (RafaelHiguera) [reply GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs]: I see 👍

File: GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs (hunk 1 above)
Comment (RafaelHiguera) [reply GooglePlayPlugins/com.google.play.billing/Samples/TrivialKart/scripts/Controller/NetworkRequestController.cs]: Yep, will do 👍

**Review by Qing451800 — APPROVED** [review]

**Review by searoadtosoftware — APPROVED** [review]

**Pull request merged**